src/threadutil.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <atomic>
+#include <thread>
+
+#include <threadutil.h>
+
+/*
+ * TODO: using thread_local changes the abi in ways that may not play nice
+ * when the c++ stdlib is linked dynamically. Disable it until thorough
+ * testing has been done.
+ *
+ * mingw32's implementation of thread_local has also been shown to behave
+ * erroneously under concurrent usage; see:
+ *
+ *   https://gist.github.com/jamesob/fe9a872051a88b2025b1aa37bfa98605
+ */
+#undef HAVE_THREAD_LOCAL
```

Comment:
I think this line may be unintentional.

Reply:
I don't think so - see the comment above; sounds like @theuni wants to do more testing with dynamic linking before we use `thread_local` outside the context of debug code.

Reply:
Unsure if we should have the code in in that case, as it's untestable (it also doesn't compile currently).